Solutions/Corvus.SpecFlow.Extensions.Specs/Containers/PerScenarioContainer.feature

```diff
@@ -0,0 +1,33 @@
+﻿# These tests are a bit meta, because they test that the library enables us to write SpecFlow specs in the way we want,
+# and they do so through the medium of SpecFlow specs. In some cases this is straightforward, but it makes certain
+# aspects of the test look a little odd.
+
+# This tag is essentially the thing we're testing. Tests using per-scenario containers will put this tag at the top of
+# their files.
+@perScenarioContainer
+
+# This tag required for these tests to work. It enables before/after bindings that perform work needed to test
+# the behaviour of the @perScenarioContainer feature.
+@runPerScenarioContainerTests
+
+Feature: PerScenarioContainer
+	When I write a test scenario that needs its own DI container
+	As a developer
+	I want to be able to define container services in a BeforeScenario binding and then use those services later in the test
+
+Background:
+	Given I have specified the perScenarioContainer tag
+
+Scenario: Services added during PopulateServiceCollection BeforeScenario phase are in Service Provider available to tests
+    Given I use scenario ContainerBindings.GetServiceProvider during a Given step
+	When I use scenario ContainerBindings.GetServiceProvider during a When step
+    Then if I also use scenario ContainerBindings.GetServiceProvider during a Then step
```

Comment:
Since lines 22-24 are essentially the "Act" parts of the spec, it would make a lot more sense to me if they were prefixed "When", "And", "And" rather than "Given", "When", "Then".

Reply:
How would that work? The whole point of this test is that I want to make sure that if you do a particular thing inside a `Given` step, it behaves in a particular way. I don't see any way of doing that without making it a `Given` step. Same for `When` and `Then`.

In a ideal world I'd be able to write:

```
When this bit of code runs in a Then step
```

but because that starts with `When` it will run as a `When` step, thus failing to test the condition it's supposed to test.

The basic problem here is that I'm attempting to test what happens when you use this code via SpecFlow in certain ways. And that sometimes necessarily means that the spec actually can't accurately describe what it is I want to test. This isn't really a spec. It's some input to SpecFlow designed to trigger certain behaviours and make certain observations.